src/datafile.rkt

```diff
@@ -29,27 +29,27 @@
                note
                tests))
 
-(define (try-list-accessor acc fail)
-  (λ (l) (if (null? l) fail (acc l))))
-
 (define (trs->pareto trs)
-  (define cas (map table-row-cost-accuracy trs))
-  (define starts (map (try-list-accessor first (list 0 0)) cas))
-  (define ptss (map (try-list-accessor (λ (ca) (cons (second ca) (third ca)))
-                                       (list (list 0 0)))
-                    cas))
-  (define reprs (map (compose get-representation table-row-precision) trs))
-
-  (define start
-    (for/fold ([x 0] [y 0] #:result (cons x y)) ([s starts])
-      (values (+ x (first s)) (+ y (second s)))))
-  (define ptss*
-    (for/list ([pts ptss])
-      (for/list ([pt pts])
-        (cons (first pt) (second pt)))))
-  (define ymax (apply + (map representation-total-bits reprs)))
-  (define frontier (map (λ (pt) (cons (first pt) (second pt))) (pareto-combine ptss #:convex? #t)))
-  (values start frontier ymax))
+  (match-let*
+      ([(cons start-cost start-accuracy)
+        (foldl (match-lambda*
+                 [(list r (cons x y))
+                  (match-let ([(list c a) ((compose first table-row-cost-accuracy) r)])
+                    (cons (+ x c) (+ y a)))])
+               (cons 0.0 0)
+               trs)]
```

Comment:
I don't see any reason to do start-cost and start-accuracy simultaneously. And if you didn't try to do them simultaneously, you could just use the normal sum operation, `(apply + values)`. I think that would be way clearer. You'd get something like:

```
(define start-cost (apply + (map (compose first first table-row-cost-accuracy) trs)))
(define start-accuracy (apply + (map (compose second first table-row-cost-accuracy) trs)))
```